packages/editor-preview/src/browser/editor-preview-manager.ts

```diff
@@ -14,151 +14,104 @@
  * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
  ********************************************************************************/
 
-import { injectable, inject, postConstruct } from '@theia/core/shared/inversify';
-import URI from '@theia/core/lib/common/uri';
-import { ApplicationShell, DockPanel } from '@theia/core/lib/browser';
 import { EditorManager, EditorOpenerOptions, EditorWidget } from '@theia/editor/lib/browser';
-import { EditorPreviewWidget } from './editor-preview-widget';
-import { EditorPreviewWidgetFactory, EditorPreviewWidgetOptions } from './editor-preview-factory';
+import { inject, injectable, postConstruct } from '@theia/core/shared/inversify';
 import { EditorPreviewPreferences } from './editor-preview-preferences';
-import { WidgetOpenHandler, WidgetOpenerOptions } from '@theia/core/lib/browser';
+import { NavigatableWidgetOptions, WidgetManager } from '@theia/core/lib/browser';
+import { DisposableCollection, MaybePromise } from '@theia/core/lib/common';
+import URI from '@theia/core/lib/common/uri';
+import { EditorPreviewWidgetFactory } from './editor-preview-widget-factory';
+import { EditorPreviewWidget } from './editor-preview-widget';
+import { FrontendApplicationStateService } from '@theia/core/lib/browser/frontend-application-state';
 
-/**
- * Opener options containing an optional preview flag.
- */
-export interface PreviewEditorOpenerOptions extends EditorOpenerOptions {
-    preview?: boolean
+export interface EditorPreviewOpenerOptions extends NavigatableWidgetOptions {
+    preview: boolean;
 }
 
-/**
- * Class for managing an editor preview widget.
- */
 @injectable()
-export class EditorPreviewManager extends WidgetOpenHandler<EditorPreviewWidget | EditorWidget> {
-
+export class EditorPreviewManager extends EditorManager {
     readonly id = EditorPreviewWidgetFactory.ID;
 
-    readonly label = 'Code Editor Preview';
-
-    protected currentEditorPreview: Promise<EditorPreviewWidget | undefined>;
-
-    @inject(EditorManager)
-    protected readonly editorManager: EditorManager;
+    @inject(WidgetManager) protected readonly widgetManager: WidgetManager;
```

Comment:
Minor: This exists on the WidgetOpenHandler already

Reply:
Good point - I've deleted it here.